File: 0169-majority-element/0169-majority-element.ts
function majorityElement(nums: number[]): number {
    
    const size = nums.length;
    let count = 0 , element : number;
    for(let i = 0; i<size; i++){
        if(count == 0){
            count++;
            element = nums[i];
        }else if(element == nums[i]) count++;
        else count--;
    }
    return element;
    
};

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int element , count = 0;
        int n = nums.size();
        for(int i = 0; i<n; i++){
            if(count == 0){
                count++;
                element = nums[i];
            }
            else if(element == nums[i]) count++;
            else count--;
        }
        return element;
    }
};